flax/linen/fp8_ops.py

```diff
@@ -664,3 +623,24 @@ def __call__(self, *args, **kwargs):
 class NANOOFp8DotGeneralOp(Fp8DotGeneralOp):
   e4m3_dtype: DType = jnp.float8_e4m3fnuz
   e5m2_dtype: DType = jnp.float8_e5m2fnuz
+
+class Fp8Einsum(Fp8DotGeneralBase):
```

Comment:
Should Fp8Einsum related to Fp8DirectDotGeneralOp? In Fp8DirectDotGeneralOp, preferred_element_type=x.dtype, shouldn't Fp8Einsum follow the same setting?

Reply:
Added support for user defined ·precision· and ·preferred_element_type·. If not provided, they will be inferred by jnp.einsum and passed to _dot_general.